gmso/formats/networkx.py

```diff
@@ -0,0 +1,953 @@
+    except AttributeError:
+        return print("Atomtypes are missing, so no bond types can be selected")
+    mia_bond_flag = 0
+    if atom_name1 and atom_name2:
+        for edge in list(networkx_graph.edges):
+            if (
+                edge[0].name == atom_name1
+                and edge[1].name == atom_name2
+                or edge[0].name == atom_name2
+                and edge[1].name == atom_name1
+            ):
+                list_of_bonds.append(
+                    edge[0].atom_type.name + " --- " + edge[1].atom_type.name
+                )
+                list_of_edges.append(edge)
+    elif atom_name1:
+        for edge in list(networkx_graph.edges):
+            if edge[0].name == atom_name1 or edge[1].name == atom_name1:
+                list_of_bonds.append(
+                    edge[0].atom_type.name + " --- " + edge[1].atom_type.name
+                )
+                list_of_edges.append(edge)
+    else:
+        for nodes in list(networkx_graph.edges.items()):
+            if nodes[1]["connection"].bond_type is None:
+                list_of_bonds.append(
+                    nodes[0][0].atom_type.name + " --- " + nodes[0][1].atom_type.name
+                )
+                list_of_edges.append((nodes[0][0], nodes[0][1]))
+                mia_bond_flag = 1
+        if not mia_bond_flag:
+            return print("All bonds are typed")
+
+    labeled_bonds = zip(list_of_bonds, list_of_edges)
+
+    # create a dic so each selected bond selects the proper edges on the networkx.graph object.
+    selectable_list = {}
+    for label, bond in labeled_bonds:
+        if label in selectable_list.keys():
+            selectable_list[label].append(bond)
+        else:
+            selectable_list[label] = []
+            selectable_list[label].append(bond)
+
+    # turn the dic selectable list into a list of tuples.
+    list_of_edges = []
+    for key in selectable_list:
+        list_of_edges.append((key, selectable_list[key]))
+
+    return list_of_edges
+
+
+def plot_networkx_bonds(
+    networkx_graph,
+    atom_name1=None,
+    atom_name2=None,
+    list_of_labels=["atom_type.name"],
+    list_of_bonds=[],
+):
+
+    fig, ax = plt.subplots(1, 1, figsize=(8, 8))
+
+    # Create dictionaries of edges that correspond red thick lines for the selected bonds
+    edge_weights = {}
+    edge_colors = {}
+    for bond in networkx_graph.edges:
+        if bond in list_of_bonds:
+            edge_weights[bond] = 5
+            edge_colors[bond] = "red"
+        else:
+            edge_weights[bond] = 1
+            edge_colors[bond] = "k"
+
+    ax = plot_networkx_nodes(
+        networkx_graph,
+        ax,
+        edge_weights=edge_weights,
+        edge_colors=edge_colors,
+        list_of_labels=list_of_labels,
+        atom_name=atom_name1,
+    )
+    return fig, ax
+
+
+def report_bond_parameters(topology, edge):
+    # return nicely printed bond parameters for a given edge.
+    for bond in list(topology.bonds):
```

Comment:
Why is the casting important here? Its a tuple right?
```
    for bond in list(topology.bonds):
```

Reply:
Fixed.